File: insurance-backend/src/main/java/co/za/insurance/company/Company.java
package co.za.insurance.company;

import java.time.LocalDateTime;
import java.util.List;

import co.za.insurance.policy.Policy;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Entity
@Getter
@Setter
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String address;
    private String city;
    private String region;
    private String postalCode;
    private String country;
    private String phoneNumber;
    private String email;
    private String website;
    private String registrationNumber;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "company", cascade = CascadeType.ALL)
    private List<Policy> policies;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: insurance-management/src/app/interfaces/company.ts
export interface Company {
  id: string;
  name: string;
  address: string;
  city: string;
  region: string;
  postalCode: string;
  country: string;
  phoneNumber?: string;
  email?: string;
  website?: string;
  registrationNumber?: string;
  createdAt: string;
  updatedAt: string;
}

// DTO for creating a new company
export interface CreateCompanyDto {
  name: string;
  address: string;
  city: string;
  region: string;
  postalCode: string;
  country: string;
  phoneNumber?: string;
  email?: string;
  website?: string;
  registrationNumber?: string;
}

// DTO for updating an existing company
export interface UpdateCompanyDto {
  name?: string;
  address?: string;
  city?: string;
  region?: string;
  postalCode?: string;
  country?: string;
  phoneNumber?: string;
  email?: string;
  website?: string;
  registrationNumber?: string;
}